client/pull.go

```diff
@@ -102,9 +110,22 @@ type Downloader struct {
 	BufferSize int64
 }
 
-// httpGetRangeRequest performs HTTP GET range request to URL specified by 'u' in range start-end.
-func (c *Client) httpGetRangeRequest(ctx context.Context, url string, start, end int64) (*http.Response, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+// httpGetRangeRequest performs HTTP GET range request to URL 'u' in range start-end.
+func (c *Client) httpGetRangeRequest(ctx context.Context, endpoint, authToken string, start, end int64) (*http.Response, error) {
+	return c.httpRangeRequest(ctx, http.MethodGet, endpoint, authToken, start, end)
+}
+
+func (c *Client) httpRangeRequest(ctx context.Context, method, endpoint, authToken string, start, end int64) (*http.Response, error) {
+	if start >= end || start < 0 || end < 0 || (end-start+1) < 0 {
```

Comment:
I don't understand what scenario `(end-start+1) < 0` is protecting against?

Is it possible for it to be true, given start >= end, and both are zero or positive?

Reply:
Removed, thanks!